doc/arch/adr-007-pa-dep-client-app.md

```diff
@@ -0,0 +1,122 @@
+
+The one major directive from the clients is that,
+from a user's perspective,
+the site should be deeply integrated with MMW.
+That is to say,
+it should use the same colors and branding,
+and not feel like its own project.
+
+From the development perspective,
+we should aim for using the latest front-end stacks,
+clean communication with the back-end API,
+and having a fast deployment pipeline.
+
+### Repository: Same vs Different
+
+Advantages of having the **same repository**:
+
+  * Familiarity, as it was done in the [Beekeepers][beekeepers] project
+  * All issues can be in one repo, simplifying project management
+  * Tasks that affect both front-end and back-end can be tracked in one commit or PR
+  * Simpler deployment, as front-end and back-end will always be in sync
+  * Styles can be easily reused between the two apps
+  * Easier to simulate production environments in development,
+    as we only need to start a single development environment.
+
+Advantages of having **different repositories**:
+
+  * Cleaner logical separation in the codebase
+  * Fewer cohabiting developer entry points, i.e. `package.json`, or multiple `node_modules`
+  * Does not require changing the `node` or `npm` version in MMW
+  * Does not require adding a Docker environment in MMW for building the front-end
+  * Faster deployment, as the client app can be deployed independently of the main app
+
+### Deployment: Paired vs Independent
+
+Advantages of **paired deployment**:
+
+  * Front-end and back-end will always be in sync
+  * Does not require extra configuration of AWS
+  * Does not need extra steps during deployment
+  * Easier to write end-to-end tests
+
+Advantages of **independent deployment**:
+
+  * Front-end can be iterated much faster, without the heavy deployment cycle of MMW
+
+### Access: Subdomain vs Subdirectory
+
+Advantages of **subdomain access**:
+
+  * More easily supports independent deployments
+
+Advantages of **subdirectory access**:
+
+  * Maintains MMW brand identity
+  * Would not have any CORS issues
+    - This would not apply to the documented public API endpoints,
+      but there are some undocumented APIs such as MapShed which we'll need to run,
+      which may have CORS headers set on them.
+
+### Option A: Same Repository, Paired Deployment, Subdirectory Access
+
+Similar to Beekeepers,
+we will create a Django app to house all files for the new client app.
+We will evaluate if the `node` version in the App VM can be upgraded,
+and if not we will use add a Docker container for building the front-end.
+Django will be configured to handle routing,
+which will render a different home page for the new routes.
+Deployments will not be affected at all and will proceed as usual.
+Some basic styles and colors will be shared between the two apps.
+
+### Option B: Different Repositories, Independent Deployments, Subdirectory Access
+
+The new app gets its own repository,
+which makes the code much easier to navigate.
```

Comment:
Having an app be its own repo feels easier to navigate for me rather than it being nestled inside a much larger app. But that perspective may be subjective and hard to defend. I'll evaluate and remove it, if needed.